File: solutions/513-find-bottom-left-tree-value/find-bottom-left-tree-value.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function findBottomLeftValue(root: TreeNode | null): number {
    const queue = [root];
    let current = null;
    while(queue.length > 0) {
        current = queue.shift();
        if(current.right) {
            queue.push(current.right);
        }
        if(current.left) {
            queue.push(current.left);
        }
    }
    return current.val;
};

File: solutions/513-find-bottom-left-tree-value/find-bottom-left-tree-value.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        result = -1

        if root is None:
            return result
        
        my_stack = [[root]]
        while len(my_stack) > 0:
            curr_level_nodes = my_stack.pop()
            next_level_nodes = []
            for node in curr_level_nodes:
                if node.left is not None:
                    next_level_nodes.append(node.left)
                if node.right is not None:
                    next_level_nodes.append(node.right)
            if len(next_level_nodes) > 0:
                my_stack.append(next_level_nodes)
            else:
                result = curr_level_nodes[0].val
        
        return result